### PR TITLE
[DNM] xapi: decode Row data during fetch

### PR DESCRIPTION
**Pull request by tiancaiamao**
`fetch` run in multiple goroutines, while `next` run in one goroutine
move the decode related code to `fetch` would improve cpu utilization

### CODE REVIEW

File: xapi/xapi.go [hunk 1]
```diff
@@ -130,66 +130,93 @@ type partialResult struct {
...
 
-// Next returns the next row of the sub result.
-// If no more row to return, data would be nil.
-func (pr *partialResult) Next() (handle int64, data []types.Datum, err error) {
-	if !pr.fetched {
-		select {
-		case err = <-pr.done:
-		}
-		pr.fetched = true
+	pr.rows = make([]selectResponseRow, len(resp.Rows))
```
Comment (shenli) [inline xapi/xapi.go]: This will call runtime.makeslice(). It cost a lot of time when there are many rows. And it also bring a lot of work for gc.

**Comment by shenli** [comment]
Any bench result？

**Comment by coocood** [comment]
LGTM

File: xapi/xapi.go [hunk 2]
```diff
@@ -130,58 +133,86 @@ type partialResult struct {
...
-	if pr.cursor >= len(pr.resp.Rows) {
-		return 0, nil, nil
+}
+
+func (pr *partialResult) decodeRows(input []*tipb.Row) ([]selectResponseRow, error) {
+	output := make([]selectResponseRow, len(input))
+	for i, row := range input {
+		handle, data, err := pr.decodeRow(row)
+		if err != nil {
+			return output, errors.Trace(err)
```
Comment (zimulala) [inline xapi/xapi.go]: output -> nil ?

**Comment by tiancaiamao** [comment]
CI would pass after https://github.com/pingcap/tidb/pull/1712

**Pull request closed without merging**